src/mpid/ch4/src/ch4r_init.c

```diff
@@ -94,12 +94,71 @@ int MPIDIG_destroy_comm(MPIR_Comm * comm)
     return mpi_errno;
 }
 
+/* Linked list internally used to keep track of
+ * allocated memory for which memory binding is
+ * requested by the user. */
+typedef struct mem_node {
+    void *ptr;
+    size_t size;
+    struct mem_node *next;
+} mem_node_t;
+
+static mem_node_t *mem_list_head = NULL;
+static mem_node_t *mem_list_tail = NULL;
+
 void *MPIDIG_mpi_alloc_mem(size_t size, MPIR_Info * info_ptr)
 {
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIG_MPI_ALLOC_MEM);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_MPI_ALLOC_MEM);
     void *p;
-    p = MPL_malloc(size, MPL_MEM_USER);
+    MPIR_hwtopo_type_e mem_type = MPIR_HWTOPO_TYPE__DDR;
+    MPIR_hwtopo_gid_t mem_gid = MPIR_HWTOPO_GID_ROOT;
+    MPIR_Info *curr_ptr = (info_ptr) ? info_ptr->next : NULL;
```

Comment:
Why is initial setting of `curr_ptr` to `info_ptr->next`?

Reply:
I guess I should just use `MPIR_Info_get_impl` instead, it's cleaner.

Reply:
Yes, that is cleaner. I'm still confused by the initialization of `curr_ptr` in `MPIR_Info_get_impl` itself, but that's a separate issue.